content/guides/end-to-end-testing/social-authentication.md

```diff
@@ -0,0 +1,422 @@
+      cy.get('[type="password"]').type(password, {
+        log: false,
+      })
+      cy.contains('Next').click().wait(4000)
+    }
+  )
+
+  cy.get('h6.dropdown-header').should('contain', name)
+}
+```
+
+And lastly, our Microsoft Login Function
+
+```js
+// cypress/support/commands.ts
+import { domain as Auth0Domain } from '../../src/auth_config.json'
+
+function logIntoMicrosoft(username: string, password: string, name: string) {
+  cy.visit('http://localhost:3000')
+  cy.get('#qsLoginBtn').click()
+
+  cy.origin(Auth0Domain, () => {
+    cy.scrollTo('bottom')
+    cy.get('form[data-provider="windowslive"]').submit()
+  })
+
+  cy.origin(
+    'login.live.com',
+    {
+      args: {
+        username,
+        password,
+      },
+    },
+    ({ username, password }) => {
+      cy.get('input[type="email"]').type(username)
+      cy.get('input[type="submit"]').click()
+      cy.get('input[type="password"]').type(password, {
+        log: false,
+      })
+      cy.get('input[type="submit"]').click()
+      cy.get('#idBtn_Back').click()
+    }
+  )
+
+  cy.get('h6.dropdown-header').should('contain', name)
+}
+```
+
+We can wire together these functions in a single command, like so:
+
+```js
+// cypress/support/commands.ts
+
+Cypress.Commands.add(
+  'loginToAuth0ViaSocial',
+  (SOCIAL_PROVIDER: 'microsoft' | 'google' | 'facebook') => {
+    const log = Cypress.log({
+      displayName: 'Social LOGIN',
+      message: [`🔐 Authenticating | ${SOCIAL_PROVIDER}`],
+      // @ts-ignore
+      autoEnd: false,
+    })
+    log.snapshot('before')
+
+    switch (SOCIAL_PROVIDER) {
+      case 'microsoft':
+        logIntoMicrosoft(
+          Cypress.env('MICROSOFT_USERNAME'),
+          Cypress.env('MICROSOFT_PASSWORD'),
+          Cypress.env('MICROSOFT_NAME')
+        )
+        break
+      case 'google':
+        logIntoGoogle(
+          Cypress.env('GOOGLE_USERNAME'),
+          Cypress.env('GOOGLE_PASSWORD'),
+          Cypress.env('GOOGLE_NAME')
+        )
+        break
+      case 'facebook':
+        logIntoFacebook(
+          Cypress.env('FACEBOOK_USERNAME'),
+          Cypress.env('FACEBOOK_PASSWORD'),
+          Cypress.env('FACEBOOK_NAME')
+        )
+        break
+      default:
+        throw new Error('no social provider configured!')
+    }
+
+    log.snapshot('after')
+    log.end()
+  }
+)
+```
+
+Now, we can use our `loginToAuth0ViaSocial` command in the test. Below is our
+test to login as a user via Auth0 and run a basic sanity check.
+
+```js
+describe('Social Logins Demo', () => {
+  beforeEach(() => {
+    // can provide Facebook, Google, or Microsoft here
+    cy.loginToAuth0ViaSocial('facebook')
+    cy.visit('http://localhost:3000/')
+  })
+
+  it('navigates to the app after navigation and displays the sample project header', () => {
+    cy.get('h1').should('contain', 'React.js Sample Project')
+  })
+})
+```
+
+Here is a sample of what all 3 should look like in order.
+
+TODO: remove watermark
```

Comment:
Need to remake this video as well as upload the videos to vimeo. The vimeo update might be done in a separate PR

Reply:
addressed in 894595edb